content/blogs/2024-02-28-dbt-or-sqlmesh.md

```diff
@@ -0,0 +1,244 @@
+
+In this blog post, we'll delve into dbt and SQLMesh frameworks and how they need an orchestration engine to really shine.
+
+
+## dbt Core shortcomings
+
+[dbt](https://www.getdbt.com/) is the go-to solution for anything regarding transformation and data modeling with SQL nowadays. It’s used by top-notch companies, [can scale quite well](https://www.getdbt.com/blog/new-dbt-cloud-features-announced-at-coalesce-2023), and has a great community of users.
+
+Still, it has some flaws:
+
+- [Users are wondering why they should references to build proper dependencies between models](https://www.reddit.com/r/dataengineering/comments/zamewl/whats_wrong_with_dbt/). Other solutions often parse the SQL queries to infer the DAG of dependancies directly.
+
+
+- While dbt's model contracts offer validation through manually defined YAML schemas, including column names and types, this approach adds an additional user burden without truly tackling the root cause of data quality issues. Change in columns or type is often done on purpose, and a failed build due to an outdated contract can be disruptive, offering limited utility beyond frustration.
+
+
+- While incremental models in dbt efficiently process new or updated data, managing scenarios with pending or required processing of specific date intervals can become challenging when using the "most recent records" approach.
+
+
+- Thoushands lines of SQL is never good - Jinja macros plus a massive pile of SQL doesn't seems a better solution to scale complex business logic. 
+
+
+- Scaling across multiple projects is difficult: having too many dbt projects in a single repository is often hard to manage. You often need to split your hundred (if not thousands) models within different teams and projects. But cross-project references is not possible in dbt Core. It has been discussed in previous [roadmap discussions](https://github.com/dbt-labs/dbt-core/blob/main/docs/roadmap/2022-08-back-for-more.md?ref=blef.fr#v15-next-year) but the final decisions has been to move it into [dbt Mesh, part of the Cloud offering](https://www.getdbt.com/product/dbt-mesh).
+
+
+- User interface is mainly about a CLI an generating a static documentation website. Web UI is only available through dbt Cloud offering.
+
+
+Adding to these main shortcomings, it's worth mentioning the analogy often made: "dbt is the Terraform for data transformation". Actually [dbt is more about jQuery than Terraform](https://www.aranke.org/dbt-jquery/). The biggest reason jQuery isn’t the de facto web framework today is because it was hard to scale to large teams.
+The same flexibility that allowed small teams to deliver value quickly left behind messes of spaghetti code for large teams to maintain.
+The introduction of React in 2013 from Facebook teams promised to address these pain points by rethinking best practices, and has been the de facto web framework ever since.
```

Comment:
```suggestion
Adding to these main shortcomings, it's worth mentioning the analogy often made: "dbt is the Terraform for data transformation". Actually [dbt is more about jQuery than Terraform](https://www.aranke.org/dbt-jquery/). The biggest reason jQuery isn’t the de facto web framework today is that it was hard to scale to large teams.

The same flexibility that allowed small teams to deliver value quickly left behind messes of spaghetti code for large teams to maintain.

The introduction of React in 2013 by Facebook teams promised to address these pain points by rethinking best practices, and it has been the de facto web framework ever since.
```